src/IslandoraContextManager.php

```diff
@@ -48,7 +48,11 @@ public function evaluateContextConditions(ContextInterface $context, array $prov
     $conditions = $context->getConditions();
 
     // Apply context to any context aware conditions.
-    $this->applyContexts($conditions, $provided);
+    // Abort if the application of contexts has been unsuccessful
+    // similarly to BlockAccessControlHandler::checkAccess().
+    if (count($conditions) > 0 && !$this->applyContexts($conditions, $provided)) {
```

Comment:
Yes, the `count($conditions) > 0 `is vital, because if the `$conditions` is empty (*) `applyContexts` returns FALSE and the `evaluateContextConditions` will return FALSE. But with `count($conditions) > 0` the rest of the code of `evaluateContextConditions` is executed and it will return TRUE. 
 \*  `!empty($conditions)` is not a replacement in this case for `count($conditions) > 0`, because $conditions is an object and not an array.

Reply:
Ah Friday 🧠 when reviewing, my bad. Looking at upstream Context looks like this is done in the `applyContexts` bit: https://www.drupal.org/project/context/issues/3177007. To be consistent with the parent would you be willing to have it mirror upstream?

Reply:
Being consistent is all I strive for! See commit [b89da47](https://github.com/Islandora/islandora/pull/931/commits/b89da473f12f60a863c87bd49278fac91cfee749)